design-documents/Remove-__sleep-and-__wakeup.md

```diff
@@ -0,0 +1,35 @@
+### Remove __sleep and __wakeup methods from our code
+#### Why?
+* Serializing objects requires classes to always consider serialization process
+and it is easy for multiple developers working on the same class to forget about it.
+* PHP serialization opens up our code to potential security issues (Remote code execution).
```

Comment:
Unserialize has been already prohibited and removed whenever possible.

Reply:
Unserialize is triggered for data stored in sessions, I've specified the case when we store objects in session